lib/marked.js

```diff
@@ -8,35 +8,55 @@
 'use strict';
 
 /**
- * Block-Level Grammar
+ * Markdown block-level element rules
+ *
+ * `heading` is for [ATX headings](http://spec.commonmark.org/0.28/#atx-headings) (#) while
+ * `lheading` is for [Setext heading](http://spec.commonmark.org/0.28/#setext-headings) (=, _)
+ *
+ * TODO: Make object with properties that can call functions.
  */
-
 var block = {
-  newline: /^\n+/,
+  blockquote: /^( {0,3}> ?(paragraph|[^\n]*)(?:\n|$))+/,
+  bullet: /(?:[*+-]|\d+\.)/,
   code: /^( {4}[^\n]+\n*)+/,
+  def: /^ {0,3}\[(label)\]: *\n? *<?([^\s>]+)>?(?:(?: +\n? *| *\n *)(title))? *(?:\n+|$)/,
   fences: noop,
-  hr: /^ {0,3}((?:- *){3,}|(?:_ *){3,}|(?:\* *){3,})(?:\n+|$)/,
   heading: /^ *(#{1,6}) *([^\n]+?) *#* *(?:\n+|$)/,
-  nptable: noop,
-  blockquote: /^( {0,3}> ?(paragraph|[^\n]*)(?:\n|$))+/,
-  list: /^( *)(bull) [\s\S]+?(?:hr|def|\n{2,}(?! )(?!\1bull )\n*|\s*$)/,
-  html: /^ *(?:comment *(?:\n|\s*$)|closed *(?:\n{2,}|\s*$)|closing *(?:\n{2,}|\s*$))/,
-  def: /^ {0,3}\[(label)\]: *\n? *<?([^\s>]+)>?(?:(?: +\n? *| *\n *)(title))? *(?:\n+|$)/,
-  table: noop,
   lheading: /^([^\n]+)\n *(=|-){2,} *(?:\n+|$)/,
+  hr: /^ {0,3}((?:- *){3,}|(?:_ *){3,}|(?:\* *){3,})(?:\n+|$)/,
+  html: /^ *(?:comment *(?:\n|\s*$)|closed *(?:\n{2,}|\s*$)|closing *(?:\n{2,}|\s*$))/,
+  item: /^( *)(bull) [^\n]*(?:\n(?!\1bull )[^\n]*)*/,
+  label: /(?:\\[\[\]]|[^\[\]])+/,
+  list: /^( *)(bull) [\s\S]+?(?:hr|def|\n{2,}(?! )(?!\1bull )\n*|\s*$)/,
+  newline: /^\n+/,
+  nptable: noop,
   paragraph: /^([^\n]+(?:\n?(?!hr|heading|lheading| {0,3}>|tag)[^\n]+)+)/,
-  text: /^[^\n]+/
+  table: noop,
+  tag: '(?!(?:'
+    + 'a|em|strong|small|s|cite|q|dfn|abbr|data|time|code'
+    + '|var|samp|kbd|sub|sup|i|b|u|mark|ruby|rt|rp|bdi|bdo'
+    + '|span|br|wbr|ins|del|img)\\b)\\w+(?!:|[^\\w\\s@]*@)\\b',
+  text: /^[^\n]+/,
+  title: /(?:"(?:\\"|[^"]|"[^"\n]*")*"|'\n?(?:[^'\n]+\n?)*'|\([^()]*\))/
 };
 
-block._label = /(?:\\[\[\]]|[^\[\]])+/;
-block._title = /(?:"(?:\\"|[^"]|"[^"\n]*")*"|'\n?(?:[^'\n]+\n?)*'|\([^()]*\))/;
+/**
+ * QUESTION: This seems like a bad construction. We define block.def as the result
+ *           of a function call. The edit() function takes a parameter named the same
+ *           as what we are defining. Why?
+ */
```

Comment:
The `block.def` defined above is really just the beginning of the construction and the rest is below.

Any other way I can think of doing it would require temp variables or repeated code.